dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/security/spring2fa/TwoFactorWebAuthenticationDetails.java

```diff
@@ -27,18 +27,39 @@
  */
 package org.hisp.dhis.security.spring2fa;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import jakarta.servlet.http.HttpServletRequest;
 import org.hisp.dhis.security.ForwardedIpAwareWebAuthenticationDetails;
 
 /**
  * @author Henning Håkonsen
  * @author Lars Helge Øverland
  */
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS)
```

Comment:
Using the class name is always risky as we then cannot change it or eve move the class in package. Isn't there anything better we can use?